5227: Build pipeline for Dapi

adrian-fitsios opened the pull request.

bjss-sean-rose committed:
Updating docker compose script to build service for dapi, blob store & file upload

spartanbear committed:
Update azure-pipelines.yml for Azure Pipelines

spartanbear committed:
Update azure-pipelines.yml for Azure Pipelines

spartanbear committed:
Update azure-pipelines.yml for Azure Pipelines

bjss-sean-rose committed:
Removing blob store from docker-compose file

spartanbear committed:
Update azure-pipelines.yml for Azure Pipelines

spartanbear committed:
Adding unit and integration test

spartanbear committed:
Adding unit and integration test

adrian-fitsios committed:
Merge branch 'development' into azureveops-build

adrian-fitsios committed:
5277: Build for Dapi

adrian-fitsios committed:
Update azure-pipelines.yml

adrian-fitsios committed:
Update azure-pipelines.yml

adrian-fitsios committed:
.net core version

adrian-fitsios committed:
use different pool

talawson05 commented on azure-pipelines.yml:
```diff
@@ -1,19 +1,72 @@
+
+- task: DockerCompose@0
+  displayName: 'Push: Dev'
+  inputs:
```
you might want a condition to stop pushing to the ACR during a pull request, overwriting the image
something like
`condition: and(succeeded(), not(eq(variables['build.reason'], 'PullRequest')))`

talawson05 replied:
or use `ne` to trim that second condition down

adrian-fitsios replied:
👍 Good idea, will do that.

talawson05 commented:
Is there a reason to only build and push to Dev, and not the other containers? Is this a change in methodology?

adrian-fitsios commented:
> Is there a reason to only build and push to Dev, and not the other containers? Is this a change in methodology?

Not that I'm aware of. I will have a chat with @roses1089 and if we actually want to push to other envs I will add that too.

adrian-fitsios committed:
5227: PR Comments

- changed 1 line scripts to be 1 line instead of multi-line
- added a conditions to stop pushing on PRs
- added build/push tasks for other envs.

talawson05 commented on azure-pipelines.yml:
```diff
@@ -1,19 +1,124 @@
+    action: 'Push services'
+
+- script: docker-compose -f "docker-compose.integration.yml" down -v
+  displayName: 'Docker-compose down'
```
Just spotted this, you always want to perform the compose down, even if the build fails. I'd suggest something like
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#succeededorfailed
or 
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#always

adrian-fitsios committed:
5227: PR Comments

- always do docker-compose down, even if the build fails / is cancelled

talawson05 reviewed (approved):
LGTM
